tests/automation.js

```diff
@@ -237,33 +237,41 @@ casper.test.begin("unit tests", 25 + gfxTests.length, function(test) {
     .thenOpen("http://localhost:8000/index.html?main=tests/isolate/TestIsolate&logLevel=info&logConsole=web,page,raw")
     .withFrame(0, function() {
         casper.waitForText("DONE", function() {
-            test.assertTextExists("I m\n" +
-                                  "I a ma\n" +
-                                  "I 3\n" +
-                                  "I ma\n" +
-                                  "I 3\n" +
-                                  "I 1 isolate\n" +
-                                  "I Isolate ID correct\n" +
-                                  "I 5\n" +
-                                  "I 6\n" +
-                                  "I 1 isolate\n" +
-                                  "I ma\n" +
-                                  "I ma\n" +
-                                  "I 3 isolates\n" +
-                                  "I 1 m1\n" +
-                                  "I 5\n" +
-                                  "I 2 m2\n" +
-                                  "I 6\n" +
-                                  "I ma\n" +
-                                  "I 1 isolate\n" +
-                                  "I Isolates terminated\n" +
-                                  "I r mar\n" +
-                                  "I 3\n" +
-                                  "I mar\n" +
-                                  "I c marc\n" +
-                                  "I 3\n" +
-                                  "I marc\n" +
-                                  "I Main isolate still running");
+          var output = this.fetchText('#raw-console');
+          var expectedOutput = ["I 0: m",
+            "I 3 0 a ma",
+            "I 1: ma",
+            "I 2: 3",
+            "I 3: 1 isolate",
+            "I 4: Isolate ID correct",
+            "I 5: 5",
+            "I 6: 6",
+            "I 7: 1 isolate",
+            "I 8: ma",
+            "I 9: ma",
+            "I 10: 3 isolates",
+            "I 6 0 2 m2",
+            "I 5 0 1 m1",
+            "I 11: ma",
+            "I 12: 1 isolate",
+            "I 13: Isolates terminated",
+            "I 3 1 r mar",
+            "I 14: mar",
+            "I 3 2 c marc",
+            "I 15: marc",
+            "I 16: Main isolate still running",
+            "I DONE",
+            ""];
+          output = output.split("\n");
+          test.assert(expectedOutput.length === output.length, "Same number of lines output.");
+          var allMatch = true;
+          for (var i = 0; i < output.length; i++) {
+            if (expectedOutput.indexOf(output[i]) === -1) {
+              allMatch = false;
+              break;
+            }
+          }
```

Comment:
This is unlikely to produce any false positives, although it's theoretically possible if the test repeats one line while skipping another. To eliminate that possibility, you could sort both arrays and test that they have the same strings at the same indexes.